lib/MFA.php

```diff
@@ -112,11 +112,11 @@ public function challengeFactor(
      * @param string $code - The verification code sent to and provided by the end user.
     */
 
-    public function verifyFactor(
+    public function verifyChallenge(
         $authenticationChallengeId,
         $code
     ) {
-        $verifyPath = "auth/factors/verify";
+        $verifyPath = "auth/challenges/{$authenticationChallengeId}/verify";
 
         if (!isset($authenticationChallengeId) || !isset($code)) {
```

Comment:
Should we move this conditional up so that we can check it before building the URL?